main.go

```diff
@@ -19,9 +19,56 @@ limitations under the License.
 package main
 
 import (
-	"fmt"
+	"flag"
+	"os"
+
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/controllers"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	// +kubebuilder:scaffold:imports
 )
 
+var (
+	myscheme = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	_ = scheme.AddToScheme(myscheme)
+	_ = v1alpha1.AddToScheme(myscheme)
+	// +kubebuilder:scaffold:scheme
+}
+
 func main() {
-	fmt.Println("hello world")
+	var metricsAddr string
+	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	flag.Parse()
+
+	ctrl.SetLogger(log.ZapLogger(true))
```

Comment:
klogr?

Reply:
klogr requires flag parsing and setting v and is overall a bad experience. We already have this exported with controller-runtime. What benefits do you see with klogr?

Reply:
I think @ncdc had suggested zap at some point in the recent past? Maybe i'm mis-remembering though.

anyway, not a huge deal. If you really want me to change it I will, i'm like a 5/10 on the caring-about-it scale.

Reply:
I wish we could use similar tools and patterns across all CAP* repos, I'm fine with either, I would just like to use one

Reply:
kubebuilder v2 uses zap by default in its scaffolding: https://github.com/kubernetes-sigs/kubebuilder/blob/fb5789a12a651e07172e3d509d71d117e37c29cd/pkg/scaffold/v2/main.go#L166

+1 to the same logger in the ecosystem. Maybe we can help everyone move to kubebuilder v2 & zap.

Reply:
the general agreement on the sig-arch mailing list (some time ago; don't have the links) was to use klog / klogr for everything.

Reply:
For this PR I'm going to stick with zap. if we need to change it for some reason later we can.